### PR TITLE
API token authentication support for VMware Cloud Director

### LINKED ISSUE #11624: VMware Cloud Director provider API Access Token authentication
### Description of the feature you would like to add / User story

As a Kubernetes cluster administrator on VMware Cloud Director provider
I would like to authenticate with API Access Token
in order to offer a secure way of authentication for automation solutions with access to its API, even in environments that enforce Multi-factor Authentication for user logins.

### Solution details

- In KubeOne Terraform example manifests: additional terraform variables for api_token and auth_type (see [here](https://registry.terraform.io/providers/vmware/vcd/latest/docs#argument-reference) for reference); customer should be able to use password or api_token authentication as he sees fit
- In KubeOne: Use the user set environment variables. Either VCD_USER and VCD_PASSWORD or VCD_API_TOKEN according to VCD_AUTH_TYPE env variable or based on what is set (see [here](https://registry.terraform.io/providers/vmware/vcd/latest/docs#argument-reference) for reference).
- In KKP: Let the user select if he/she wants to provide “User Credentials” or “Application (or API Token) Credentials” during Cluster creation and preset definition like you already do for OpenStack.

### Alternative approaches

- Username and Password authentication

### Use cases

- Passwords expire, API tokens do not expire.
- Still usable when 2FA is enforced.
- Terraform, KubeOne, KKP, CSI, CCM can be connected with an API key

### Additional information

Adding it to:

- machine-controller cloud-config
- KubeOne cloud-config
- KubeOne required environment variables
- KKP cluster crd, dashboard, cloud provider

It is important to use the https://site.cloud.example.com/oauth/tenant/tenant_name/token API endpoint

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds support for API token authentication for VMware Cloud Director.

**Which issue(s) this PR fixes**:

Fixes #11624

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
VMware Cloud Director now supports authentication using API Token
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/crd/k8c.io/kubermatic.k8c.io_presets.yaml [hunk 1]
```diff
@@ -399,6 +401,7 @@ spec:
                     vdc:
                       type: string
                   required:
+                    - apiToken
```
Comment (xrstf) [inline pkg/crd/k8c.io/kubermatic.k8c.io_presets.yaml]: Is this truly a new mandatory field?

Comment (ahmedwaleedmalik) [reply pkg/crd/k8c.io/kubermatic.k8c.io_presets.yaml]: That was an oversight, fixed it. 🙇‍♂️

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**